homeassistant/helpers/entity.py

```diff
@@ -253,8 +254,35 @@ def async_update_ha_state(self, force_refresh=False):
                             'https://goo.gl/Nvioub', self.entity_id,
                             end - start)
 
+        def get_overrides(entity_id: str) -> Dict:
+            """Return a dictionary of overrides related to entity_id.
+
+            Whole-domain overrides are of lowest priorities,
+            then regexp on entity ID, and finally exact entity_id
+            matches are of highest priority.
+            """
+            domain_result = regexp_result = exact_result = {}
+            domain = split_entity_id(entity_id)[0]
+            for key, value in _OVERWRITE.items():
+                key_list = [
+                    single_key.strip() for single_key in key.split(',')]
+                if domain in key_list:
+                    domain_result = value
```

Comment:
I would recommend doing 
```python
domain_result.update(value)
exact_result.update(value)
# and
regexp_result.update(value)
```
for all three of these. This would handle the edge case where the user may have applied customizations more than once at any of the three levels. This is especially important for the regular expression matching.